lib/griddler/email.rb

```diff
@@ -26,6 +26,10 @@ def initialize(params)
       @raw_headers = params[:headers]
 
       @attachments = params[:attachments]
+
+      @spam_report = params[:spam_report]
+
+      @spam_score = extract_spam_score
```

Comment:
Is memoizing this needed? `extract_spam_score` could be renamed `spam_score`, remove the `attr_reader` and ivar, and just read from the `@spam_report` hash.